llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -88,6 +89,31 @@ void llvm::getSYCLDeviceRequirements(
   if (ReqdWorkGroupSize)
     Requirements["reqd_work_group_size_uint64_t"] = *ReqdWorkGroupSize;
 
+  auto ExtractStringFromMDNodeOperand =
+      [=](const MDNode *N, unsigned OpNo) -> llvm::SmallString<256> {
+    MDString *S = dyn_cast<llvm::MDString>(N->getOperand(OpNo).get());
```

Comment:
```suggestion
    MDString *S = cast<llvm::MDString>(N->getOperand(OpNo).get());
```

If you are not doing `nullptr` check, then you should just use `cast`. `dyn_cast` contains runtime `if (isa<>)`, whilst `cast` only has `assert(isa<>)`